### PR TITLE
Include lb and session affinity related parameters to kpng controller

### PR DESCRIPTION
**Pull request by VivekThrivikraman-est**

### CODE REVIEW

File: pkg/api/localnetv1/services.proto [hunk 1]
```diff
@@ -82,6 +82,15 @@ message Service {
     repeated PortMapping Ports = 6;
 
     bool ExternalTrafficToLocal = 7;
+
+    string SessionAffinity = 10;
```
Comment (mcluseau) [inline pkg/api/localnetv1/services.proto:86]: remove from here and move to a `SessionAffinity` message...

File: pkg/api/localnetv1/services.proto [hunk 2]
```diff
@@ -82,6 +82,15 @@ message Service {
...
+    SessionAffinityConfig SessionAffinityConfig = 11;
```
Comment (mcluseau) [inline pkg/api/localnetv1/services.proto:87]: ... rename to SessionAffinity...

File: pkg/api/localnetv1/services.proto [hunk 3]
```diff
@@ -82,6 +82,15 @@ message Service {
...
+    IPSet LoadBalancerSourceRanges = 12;
+    IPSet LoadBalancerIngressIPs = 13;
+}
+
+message SessionAffinityConfig {
```
Comment (mcluseau) [inline pkg/api/localnetv1/services.proto:92]: rename to SessionAffinity and add `string Kind = 1` that will receive the k8s `SessionAffinity` value

**Review by mcluseau — CHANGES_REQUESTED** [review]

**Comment by qabpea** [comment]
Session affinity is potentially tricky. There are situations where it does not scale very well, at L4 this typically occurs when there is L4 non transparent proxy or NAT in the way. I can see why you would do this with external IPs, but on the inside  of the cluster there is a potential problem when SNATing a node. The affinity must be set before the SNAT occur, if not then all sessions from one node will be directed towards one POD. This is currently not a problem, but it can/will become a problem if the order of LB and SNAT is not done in the correct order (or even better, never use SNAT for cluster internal traffic)

**Comment by jayunit100** [comment]
fixes #93 #64

**Comment by mcluseau** [comment]
plz test with latest master

**Comment by jayunit100** [comment]
all done now by mikael

**Pull request closed without merging**